src/file/FileOperationTrait.php

```diff
@@ -15,30 +15,23 @@
 use Stringable;
 
 trait FileOperationTrait {
-
-	/** @var string */
 	protected string $pathname;
 
-	/**
-	 * Initialize the property by converting Stringable into string.
-	 *
-	 * @param string|Stringable $pathname
-	 */
-	abstract public function __construct(string | Stringable $pathname);
-
 	abstract public function delete(): void;
 
 	/**
 	 * Static instantiator
 	 *
 	 * @param string|Stringable $pathname
 	 *
-	 * @return static
-	 *
-	 * @psalm-suppress UnsafeInstantiation
+	 * @return self
 	 */
-	public static function create(Stringable | string $pathname): static {
-		return new static($pathname);
+	public static function create(string|Stringable $pathname): self {
+		return new self($pathname);
+	}
+
+	final public function __construct(string|Stringable $filename) {
```

Comment:
Dunno, if this is a good idea to move the constructor to a trait... 🤔

Reply:
No, it isn't. I was experimenting and I forgot to rollback it.